packages/kit/src/components/ComponentForm/ComponentForm.component.js

```diff
@@ -0,0 +1,186 @@
+	static ON_SUBMIT = 'TCOMP_FORM_SUBMIT';
+	static ON_DEFINITION_URL_CHANGED = 'TCOMP_FORM_DEFINITION_URL_CHANGE';
+
+	constructor(props) {
+		super(props);
+		this.state = {};
+		this.onTrigger = this.onTrigger.bind(this);
+		this.onChange = this.onChange.bind(this);
+		this.onSubmit = this.onSubmit.bind(this);
+		this.getUISpec = this.getUISpec.bind(this);
+		this.setupTrigger = this.setupTrigger.bind(this);
+		this.getTriggers = this.getTriggers.bind(this);
+		this.setupTrigger(props);
+	}
+
+	componentDidUpdate(props) {
+		if (
+			props.triggerURL !== this.props.triggerURL ||
+			props.customTriggers !== this.props.customTriggers
+		) {
+			this.trigger = this.setupTrigger(this.props);
+		}
+		if (this.props.definitionURL !== props.definitionURL) {
+			this.props.dispatch({
+				type: ComponentForm.ON_DEFINITION_URL_CHANGED,
+				...this.props,
+			});
+		}
+	}
+
+	onChange(event, data) {
+		event.persist();
+		if (!this.props.state.get('dirty')) {
+			this.props.setState({ dirty: true });
+		}
+		const properties = data.properties;
+		if (data.schema.titleMap && data.value) {
+			// Here we add a field side by side with the value
+			// to keep the title associated to the value
+			const info = data.schema.titleMap.find(titleMap => titleMap.value === data.value);
+			let currentProp = properties;
+			let currentKey;
+			data.schema.key.forEach((key, index) => {
+				if (index !== data.schema.key.length - 1) {
+					currentProp = currentProp[key];
+				} else {
+					currentKey = key;
+				}
+			});
+			currentProp[`$${currentKey}_name`] = info.name;
+		}
+		this.setState({ properties });
+		if (this.props.dispatchOnChange) {
+			this.props.dispatch({
+				type: ComponentForm.ON_CHANGE,
+				event: {
+					type: 'onChange',
+					component: 'TCompForm',
+					props: this.props,
+					state: this.state,
+					source: event,
+				},
+				data,
+				properties,
+				uiSpec: this.getUISpec(),
+			});
+		}
+	}
+
+	onTrigger(event, payload) {
+		return this.trigger(event, payload).then(data => {
+			if (data.properties) {
+				this.setState({ properties: data.properties });
+			}
+			if (data.errors || data.jsonSchema || data.uiSchema) {
+				this.props.setState(data);
+			}
+			this.props.dispatch({
+				type: ComponentForm.ON_TRIGGER,
+				event: {
+					type: 'onTrigger',
+					component: ComponentForm,
+					componentId: this.props.componentId,
+					props: this.props,
+					state: this.state,
+				},
+				data,
+				uiSpec: this.getUISpec(),
+			});
+		});
+	}
+
+	onSubmit(event, data) {
+		event.persist();
+		if (this.props.onSubmit) {
+			this.props.onSubmit(event, data);
+		}
+		this.props.dispatch({
+			type: ComponentForm.ON_SUBMIT,
+			event,
+			...this.getUISpec(),
+			properties: data,
+		});
+	}
+
+	setupTrigger(props) {
+		this.trigger = kit.createTriggers({
+			url: props.triggerURL,
+			customRegistry: this.getTriggers(),
+		});
+	}
+
+	getTriggers() {
+		return {
+			reloadForm: ({ body }) => ({
```

Comment:
Can you explain this `reloadForm` trigger ? I am not sure to understand.
I saw that you have it in your example in datalist onFocus event. Why do we reload all the form ? That was how tcomp v0 trigger works and we wanted to get out of this. Shouldn't we have a trigger that only fetch the suggestions and reconcile the schema ? 

Plus the supported triggers should be in `kit` instead of the form